src/awscr-s3/client.cr

```diff
@@ -296,5 +296,21 @@ module Awscr::S3
     private def http
       @http
     end
+
+    private def encode(object)
+      String.build do |io|
+        URI.encode(object, io, space_to_plus: true) do |byte|
```

Comment:
`URI.encode_www_form` changes spaces to "+" symbols and the PR adding it claimed it worked but didn't add any tests showing it handling spaces. I went with what seems to work https://github.com/taylorfinnell/awscr-s3/pull/73